dafni_cli/datasets/dataset_metadata.py

```diff
@@ -76,7 +77,7 @@ class DatasetMetadata:
 
     Methods:
         __init__(): DatasetMetadata constructor
-        set_attributes_from_dict(jwt (str), dataset (dict)): Sets the dataset metadata attributes from given client model dict
+        set_attributes_from_dict(session (DAFNISession), dataset (dict)): Sets the dataset metadata attributes from given client model dict
```

Comment:
Yeah, personally I would rather just remove these sections – its only repeating what methods are defined below and hasn’t been touched for 2 years and its just another place to update the comments unless there is some reason we would want to keep them? These changes came from a bulk find & replace so I didn't really touch it myself.

Reply:
I believe we follow this standard https://peps.python.org/pep-0257/ which includes listing public class methods in the docstring

Reply:
I have updated it for this PR anyway though (although I haven't added any missing ones if there are any) - I can address this properly in another PR if we want to keep them or otherwise remove them.

Reply:
Ah yes I see, the specific part seems to be
"The docstring for a class should summarize its behavior and list the public methods and instance variables. If the class is intended to be subclassed, and has an additional interface for subclasses, this interface should be listed separately (in the docstring). The class constructor should be documented in the docstring for its __init__ method. Individual methods should be documented by their own docstring."

I did find https://stackoverflow.com/questions/62444612/should-i-list-class-methods-in-the-class-docstring, I cant see any reference to it explicitly being like we are using here with full repeated descriptions and parameter definitions, the sample from datetime only lists the function names and ignores the arguments. It seems that most ignore it entirely. I must admit I don't see why its a standard since vscode will just show the docstring when you look at the methods, but then it was written in 2001.